fix(schemas): Deploy Schemas build fails in CI with "input device is not a TTY"

## Problem

The first real `Deploy Schemas` run on a schemas version bump (v3002, run [27574386275](https://github.com/mozilla/experimenter/actions/runs/27574386275)) failed at the **Build schemas** step:

```
docker run --rm -ti  -v ./schemas:/schemas ... schemas:dev sh -c "poetry run python generate_json_schema.py && poetry build"
the input device is not a TTY
make: *** [Makefile:354: schemas_build] Error 1
```

The PyPI and npm publish steps were then skipped, so nothing published.

## Root cause

`make`'s `SCHEMAS_RUN` uses `DOCKER_RUN_INTERACTIVE`, which defaults to `-ti` (`Makefile:25`). GitHub Actions runners have no TTY, so `docker run -t` fails immediately. `check-schemas.yml` avoids this because it goes through the `setup-cached-build` action, which sets `DOCKER_RUN_INTERACTIVE=` in the environment. `deploy-schemas.yml` runs `make schemas_build` directly and never clears the flag.

This is a latent bug exposed by the first real schemas release to run through the GHA deploy — all prior Deploy Schemas runs were skip-runs on non-schemas pushes. Unrelated to the recent Trusted Publishing changes (#15896, #15931), whose publish steps never executed.

## Fix

Set `DOCKER_RUN_INTERACTIVE: ""` as a job-level env in `deploy-schemas.yml` (mirrors what `setup-cached-build` does for the check workflow). An empty env var defeats the Makefile `?=` default, so `docker run` is invoked without `-t`.

## Acceptance criteria

- `Deploy Schemas` builds the schemas package in CI without the TTY error.
- A subsequent schemas version bump publishes to both PyPI and npm.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-7009)

--- a/.github/workflows/deploy-schemas.yml
+++ b/.github/workflows/deploy-schemas.yml
@@ -10,6 +10,8 @@ jobs:
   deploy:
     name: Deploy Schemas
     runs-on: ubuntu-24.04
+    env:
+      DOCKER_RUN_INTERACTIVE: ""
     permissions:
       contents: read
       id-token: write

